app/src/main/java/com/example/themoviedb/loginvalidation/LoginViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.themoviedb.loginvalidation
+
+import android.widget.TextView
+import android.widget.Toast
+import androidx.lifecycle.ViewModel
+import com.example.themoviedb.application.MyApp
+import com.example.themoviedb.application.PASSWORD_KEY
+import com.example.themoviedb.application.USERNAME_KEY
+
+class LoginViewModel(private val repository: SharedPreferencesRepository) : ViewModel(), LoginViewModelInterface {
+
+
+    override fun verifyCredentials(username: TextView, password: TextView): Boolean {
```

Comment:
Also,on what Thread is this being called? Remember,this counts as a "long running" process